UnityWeld_Editor/AnimatorParameterBindingEditor.cs

```diff
@@ -0,0 +1,343 @@
+
+            EditorGUILayout.Space();
+
+        }
+
+        private void ShowAnimatorParametersMenu(
+            GUIContent label,
+            Action<AnimatorParameterTypeAndName> propertyValueSetter,
+            AnimatorParameterTypeAndName curPropertyValue,
+            AnimatorControllerParameter[] properties,
+            out Type selectedPropertyType
+        )
+        {
+            if(properties == null || properties.Count() <= 0)
+            {
+                selectedPropertyType = null;
+                return;
+            }
+
+            var propertyNamesAndTypes = properties
+                .Select(m => new AnimatorParameterTypeAndName(m.name, m.type))
+                .ToArray();
+            var selectedIndex = Array.IndexOf(propertyNamesAndTypes, curPropertyValue);
+            var content = properties.Select(prop => new GUIContent(string.Concat(
+                    prop.name,
+                    " : ",
+                    prop.type.ToString()
+                )))
+                .ToArray();
+
+            var newSelectedIndex = EditorGUILayout.Popup(label, selectedIndex, content);
+            if (newSelectedIndex != selectedIndex)
+            {
+                var newSelectedProperty = properties[newSelectedIndex];
+
+                UpdateProperty(
+                    propertyValueSetter,
+                    curPropertyValue,
+                    new AnimatorParameterTypeAndName(newSelectedProperty.name, newSelectedProperty.type),
+                    "Set Animator parameter"
+                );
+
+                selectedPropertyType = AnimatorControllerParameterTypeToType(newSelectedProperty.type);
+            }
+            else
+            {
+                if (selectedIndex < 0)
+                {
+                    selectedPropertyType = null;
+                    return;
+                }
+
+                selectedPropertyType = AnimatorControllerParameterTypeToType(properties[selectedIndex].type);
+            }
+        }
+
+        /// <summary>
+        /// Returns the corresponding System.Type from AnimatorControllerParameterType
+        /// </summary>
+        /// <param name="parameterType">The type of parameter</param>
+        /// <returns>The System.Type corresponding to paramter type</returns>
+        private Type AnimatorControllerParameterTypeToType(AnimatorControllerParameterType parameterType)
+        {
+            switch (parameterType)
+            {
+                case AnimatorControllerParameterType.Bool:
+                        return typeof(bool);
+                case AnimatorControllerParameterType.Float:
+                        return typeof(float);
+                case AnimatorControllerParameterType.Int:
+                        return typeof(int);
+                case AnimatorControllerParameterType.Trigger:
+                        return typeof(AnimatorParameterTrigger);
+                default:
+                        return null;
```

Comment:
Maybe instead of returning null this could throw an ArgumentOutOfRangeException. You have accounted for all the types of AnimatorControllerParameterTypes that exist in Unity at the moment but it is still good to account for the possibility that in the future they may introduce another one. If that ever happened, I think having an exception thrown here with an appropriate message would make more sense than just returning null, which could cause a NullReferenceException somewhere else unexpected in the code.